src/parseTree.js

```diff
@@ -18,11 +18,13 @@ const parseTree = function(toParse: string, settings: Settings): ParseNode<*>[]
         throw new TypeError('KaTeX can only parse string typed expression');
     }
     const parser = new Parser(toParse, settings);
+    // Blank out any \df@tag to avoid spurious "Duplicate \tag" errors
+    delete parser.gullet.macros.current["\\df@tag"];
```

Comment:
I think `current` should not be accessed directly, and it'd be nice to have expression-scope namespace as we may add other state-macros like `\df@tag`.

Reply:
How does LaTeX handle this?

Reply:
@ylemkimon is right!  I tested the following with LaTeX:
```latex
\def\x{1}\x
\[\def\x{2}\]
\x
$\def\x{3}$
\x
$$\def\x{4}$$
\x
```
and all of the `\x`s (outside math mode where they're set) evaluate to `1`.  So TeX does make a group for each math-mode expression, justifying adding such a thing.